src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,120 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_INDEX = 10;
+    private static final double GROW_INDEX = 1.5;
+    private int maxSize = DEFAULT_INDEX;
+    private int sizeOfArray;
+    private T[] arrayDefault;
+    private T[] arrayTemporary;
```

Comment:
```suggestion
    private T[] array;
```
temporary array should be local variable